pinot-core/src/main/java/org/apache/pinot/core/operator/query/SelectionOrderByOperator.java

```diff
@@ -127,14 +125,22 @@ private Comparator<Object[]> getComparator() {
     // Compare all single-value columns
     int numOrderByExpressions = _orderByExpressions.size();
     List<Integer> valueIndexList = new ArrayList<>(numOrderByExpressions);
+    List<OrderByExpressionContext> mvOrderByExpressions = new ArrayList<>();
     for (int i = 0; i < numOrderByExpressions; i++) {
       if (_orderByExpressionMetadata[i].isSingleValue()) {
         valueIndexList.add(i);
       } else {
-        _queryHasMVSelectionOrderBy = true;
+        mvOrderByExpressions.add(_orderByExpressions.get(i));
```

Comment:
(minor) We may directly throw exception here to have slightly better performance. IMO no need to gather all MV order-by expressions before throwing

Reply:
done